### ISSUE TITLE
okhttp fails to build when complies with Java version 1.6 for android

### ISSUE THREAD

**Issue by lhwa** [square/okhttp#528]
Set JAVA_HOME to use jdk1.6 and run "mvn compile" on the latest master branch, then get the following build errors.  This is an issue since this makes okhttp unable to build for android apps where jdk1.6 is required. 

[INFO] --- maven-compiler-plugin:3.0:compile (default-compile) @ okhttp ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 44 source files to /Users/leehwa/Programming/ThirdParty/okhttp/okhttp/target/classes
[INFO] -------------------------------------------------------------
[WARNING] COMPILATION WARNING :
[INFO] -------------------------------------------------------------
[WARNING] Note: Some input files use or override a deprecated API.
[WARNING] Note: Recompile with -Xlint:deprecation for details.
[INFO] 2 warnings
[INFO] -------------------------------------------------------------
[INFO] -------------------------------------------------------------
[ERROR] COMPILATION ERROR :
[INFO] -------------------------------------------------------------
[ERROR] /Users/leehwa/Programming/ThirdParty/okhttp/okhttp/src/main/java/com/squareup/okhttp/internal/http/HttpsURLConnectionImpl.java:[304,3] method does not override or implement a method from a supertype
[ERROR] /Users/leehwa/Programming/ThirdParty/okhttp/okhttp/src/main/java/com/squareup/okhttp/internal/http/HttpsURLConnectionImpl.java:[309,20] cannot find symbol
symbol  : method getContentLengthLong()
location: class com.squareup.okhttp.internal.http.HttpURLConnectionImpl
[ERROR] /Users/leehwa/Programming/ThirdParty/okhttp/okhttp/src/main/java/com/squareup/okhttp/internal/http/HttpsURLConnectionImpl.java:[308,3] method does not override or implement a method from a supertype
[ERROR] /Users/leehwa/Programming/ThirdParty/okhttp/okhttp/src/main/java/com/squareup/okhttp/internal/http/HttpsURLConnectionImpl.java:[313,20] cannot find symbol
symbol  : method getHeaderFieldLong(java.lang.String,long)
location: class com.squareup.okhttp.internal.http.HttpURLConnectionImpl
[ERROR] /Users/leehwa/Programming/ThirdParty/okhttp/okhttp/src/main/java/com/squareup/okhttp/internal/http/HttpsURLConnectionImpl.java:[312,3] method does not override or implement a method from a supertype
[INFO] 5 errors
[INFO] -------------------------------------------------------------
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO]
[INFO] OkHttp (Parent) ................................... SUCCESS [5.224s]
[INFO] OkHttp SPDY and HTTP/2.0 internals ................ SUCCESS [10.657s]
[INFO] MockWebServer ..................................... SUCCESS [2.771s]
[INFO] OkHttp ............................................ FAILURE [0.401s]
[INFO] OkHttp Apache HttpClient .......................... SKIPPED
[INFO] OkCurl ............................................ SKIPPED
[INFO] Samples (Parent) .................................. SKIPPED
[INFO] Sample: Guide ..................................... SKIPPED
[INFO] Sample: Simple Client ............................. SKIPPED
[INFO] Sample: Static Server ............................. SKIPPED
[INFO] Benchmarks ........................................ SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 19.676s
[INFO] Finished at: Thu Feb 13 13:16:06 PST 2014
[INFO] Final Memory: 14M/81M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.0:compile (default-compile) on project okhttp: Compilation failure: Compilation failure:
[ERROR] /Users/leehwa/Programming/ThirdParty/okhttp/okhttp/src/main/java/com/squareup/okhttp/internal/http/HttpsURLConnectionImpl.java:[304,3] method does not override or implement a method from a supertype
[ERROR] /Users/leehwa/Programming/ThirdParty/okhttp/okhttp/src/main/java/com/squareup/okhttp/internal/http/HttpsURLConnectionImpl.java:[309,20] cannot find symbol
[ERROR] symbol  : method getContentLengthLong()
[ERROR] location: class com.squareup.okhttp.internal.http.HttpURLConnectionImpl
[ERROR] /Users/leehwa/Programming/ThirdParty/okhttp/okhttp/src/main/java/com/squareup/okhttp/internal/http/HttpsURLConnectionImpl.java:[308,3] method does not override or implement a method from a supertype
[ERROR] /Users/leehwa/Programming/ThirdParty/okhttp/okhttp/src/main/java/com/squareup/okhttp/internal/http/HttpsURLConnectionImpl.java:[313,20] cannot find symbol
[ERROR] symbol  : method getHeaderFieldLong(java.lang.String,long)
[ERROR] location: class com.squareup.okhttp.internal.http.HttpURLConnectionImpl
[ERROR] /Users/leehwa/Programming/ThirdParty/okhttp/okhttp/src/main/java/com/squareup/okhttp/internal/http/HttpsURLConnectionImpl.java:[312,3] method does not override or implement a method from a supertype
[ERROR] -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
[ERROR]
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <goals> -rf :okhttp

**Comment by ItamarM** [comment]
Hi @monteiz I forked the okhttp to support Java 6 https://github.com/Nextpeer/okhttp 
Take a look if it still bugs you.

**Comment by monteiz** [comment]
Thanks @ItamarM but at the moment I do not use okhttp.

Cheers

**Comment by bluebery** [comment]
I also required a java 1.6 build of okhttp but I needed an updated one for v2.3 so I forked @ItamarM again to bring the repo up to date with square

Furthermore, I had to fork directly from square/okio to generate a java 1.6 friendly build of okio (1.3.0) required by okhttp.

https://github.com/bluebery/okhttp
https://github.com/bluebery/okio

**Comment by xhh** [comment]
The [Building wiki page](https://github.com/square/okhttp/wiki/Building) says

> Runtime compatibility with Java 6 is enforced as part of the build to ensure compliance with Android and older versions of the JVM.

so Java 6 is supported in runtime, right? But the [home page] says:

> OkHttp supports Android 2.3 and above. For Java, the minimum requirement is 1.7.

which one is correct?

**Comment by swankjesse** [comment]
We require Java 7 or Android 2.3. The building page is out of date.

**Comment by cameron-gagnon** [comment]
@bluebery I cloned your two repos and opened them up in IntelliJ. When I went to build the whole project after importing it, many of the files still failed to build. It turns out that the `okhttp/okhttp` part of the library will build fine for Java 1.6, however many of the other modules will not, due to the diamond operator. It may be worth clearing this up in your `README.md` so others don't waste time, or completely bypass your repo if all they need is the `okhttp/okhttp` module to be Java 1.6 compatible. I plan to submit a PR with many of the other modules fixed up with the Java 1.6 compatible use of the diamond operator.

I also wanted to drop this note here for anyone else looking for a Java 1.6 compliant version of this library (or just a part of it) as I was.
